analysis_test.go

```diff
@@ -1,62 +1,57 @@
 package main
 
 import (
-	"bytes"
 	"errors"
-	"io"
-	"io/ioutil"
 	"testing"
 )
 
-var b *bytes.Buffer
-
-func mockOutFor(project, resource string) (io.Writer, io.Closer, error) {
-	return b, ioutil.NopCloser(nil), nil
-}
-
 func mockCheckFactoryOnePassOneFail() []CheckTask {
 	return []CheckTask{mockTestOne, mockTestTwo}
 }
 func mockCheckFactoryOnePass() []CheckTask {
 	return []CheckTask{mockTestOne}
 }
 
-func mockTestOne(project string, stdErr io.Writer) (Result, error) {
+func mockTestOne(project, basepath string) (Result, error) {
 	result := Result{StatusMessage: "Called mockTestOne"}
 	return result, nil
-
 }
 
-func mockTestTwo(project string, stdErr io.Writer) (Result, error) {
+func mockTestTwo(project, basepath string) (Result, error) {
 	result := Result{StatusMessage: "Called mockTestTwo"}
 	return result, errors.New("FAIL")
 }
 
 func TestCheckTasks(t *testing.T) {
-	b = bytes.NewBuffer([]byte{})
-	task := checkTasks(mockCheckFactoryOnePassOneFail, "MockProject", mockOutFor, mockOutFor)
+	results := make(chan CheckResults, 2)
+	task := checkProjectTask(mockCheckFactoryOnePassOneFail, "MockProject", "/", results)
 	err := task()
 
 	if err == nil {
 		t.Fatal("Expected error")
 	}
 
-	if len(string(b.Bytes())) == 0 {
-		t.Fatal("Tests not called")
+	close(results)
 
+	for result := range results {
+		if (len(result.Results)) != 2 {
+			t.Fatal("Expected 2 check results, got: " + string(len(result.Results)))
+		}
 	}
 
-	b = bytes.NewBuffer([]byte{})
-	task = checkTasks(mockCheckFactoryOnePass, "MockProject", mockOutFor, mockOutFor)
+	results = make(chan CheckResults, 1)
+	task = checkProjectTask(mockCheckFactoryOnePass, "MockProject", "/", results)
 	err = task()
 
 	if err != nil {
 		t.Fatal("Expected no errors")
 	}
 
-	if len(string(b.Bytes())) == 0 {
-		t.Fatal("Tests not called")
+	close(results)
```

Comment:
As this is a unit test - if nothing is sent on the channel then the test should fail.

I can assert that I should have received something on this channel by the time I close it, as the code that sends into this channel runs synchronously, at line 28.